packages/core/src/state-management/async-lru-map.ts

```diff
@@ -0,0 +1,197 @@
+import { NamedPQueue } from './named-p-queue';
+
+const NEWER = Symbol('newer');
+const OLDER = Symbol('older');
+
+class Entry<A> {
+  [NEWER]: Entry<A> | undefined;
+  [OLDER]: Entry<A> | undefined;
+
+  constructor(readonly key: string, public value: A) {}
+}
+
+const noop = async () => {
+  // Do Nothing
+};
+
+/**
+ * LRU Map with add/set/delete operations linearized per key.
```

Comment:
I'm finding this class complicated and quite difficult to decipher.

I'm also wondering if it's trying to do a lot more than it actually needs to.  For one, why do we need the linearization per key?  Isn't there only ever going to be one element per key anyway?  Can we remove all concurrency control from this class and do it at a higher level?

Also, the code here seems to be heavily based on [lru-map](https://github.com/rsms/js-lru/blob/master/lru.js).  Is there any reason we couldn't just use that package directly?

Reply:
Sounds like this will go away in a future version, so not gonna worry about it right now